ludwig/visualize.py

```diff
@@ -147,6 +151,74 @@ def _validate_output_feature_name_from_test_stats(
     except TypeError:
         return output_feature_names_set
 
+def _encode_categorical_feature(
+        raw: np.array,
+        str2idx: dict
+) -> np.array:
+    """encodes raw categorical string value to encoded numeric value
+
+    Args:
+    :param raw: (np.array) string categorical representation
+    :param str2idx: (dict) dictionary that maps string representation to
+        encoded value.
+
+    Returns:
+        np.array
+    """
+    return str2idx[raw]
+
+def _extract_ground_truth_values(
+        ground_truth: str,
+        output_feature_name: str,
+        ground_truth_split: int,
+        split_file: Union[str, None]) -> pd.Series:
+    """Helper function to extract ground truth values
+
+    Args:
+    :param ground_truth: (str) path to source data containing ground truth.
+    :param output_feature_name: (str) output feature name for ground
+        truth values.
+    :param ground_truth_split: (int) dataset split to use for ground truth,
+        defaults to 2.
+    :param split_file: (Union[str, None]) optional file path to split values.
+
+    # Return
+
+    :return pd.Series: ground truth values from source data set
+
+    """
+    # determine ground truth data format and get appropriate reader
+    data_format = figure_data_format_dataset(ground_truth)
+    if data_format not in CACHEABLE_FORMATS:
```

Comment:
I would say instead of returning an error, if the dataset is non cacheable (meaning it-s a DF) just do `gt_df = ground_truth`